math/src/field/element.rs

```diff
@@ -92,7 +94,81 @@ where
     }
 }
 
-/// AddAssign operator overloading for field elements
+impl<F> Shl<usize> for FieldElement<F>
+where
+    F: IsLinearField,
```

Comment:
We implement operations `<<`, `>>`, and `&` for `FieldElement` only if `F` is a `IsLinearField`.
(You'll see later that `IsLinearField : HasFieldOperations`).